pygfx/utils/transform.py

```diff
@@ -0,0 +1,212 @@
+import numpy as np
+import pylinalg as pla
+from time import perf_counter_ns
+
+from typing import List, Tuple
+
+
+class cached:
+    """Cache for computed properties.
+
+    This descriptor implements a minimal timestamp-based cache for computed
+    properties. The value of the property is computed using ``compute_fn`` and
+    the result is cached until ``obj.last_modified`` advances. At this point the
+    value of the computed property is recomputed upon the next read.
+
+    """
+
+    def __init__(self, compute_fn=None) -> None:
+        self.compute_fn = compute_fn
+        self.last_updated = None
+        self.cached_value = None
+
+    def __get__(self, obj: "AffineBase", objtype=None):
+        if obj is None:
+            return self
+
+        if self.last_updated is None or obj.last_modified > self.last_updated:
```

Comment:
Sure. I wrote it this way to not assume what `obj.last_modified` actually is (timestamp, int, custom object, ...) as we were still discussing its type and I wanted an easy way to update.

Since we settled for `perf_counter_ns()` we can initialize with 0.